src/test/shared/ui/input.test.ts

```diff
@@ -0,0 +1,281 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import * as vscode from 'vscode'
+import * as input from '../../../shared/ui/input'
+
+describe('createInputBox', async () => {
+    let testInput: vscode.InputBox | undefined
+
+    afterEach(() => {
+        if (testInput) {
+            testInput.dispose()
+        }
+    })
+
+    it('Sets buttons', async () => {
+        const buttons: vscode.QuickInputButton[] = [
+            vscode.QuickInputButtons.Back
+        ]
+
+        testInput = input.createInputBox({
+            buttons: buttons
+        })
+
+        assert.deepStrictEqual(testInput.buttons, buttons)
+    })
+
+    it('Sets Options', async () => {
+        const options = {
+            title: 'title',
+            placeHolder: 'placeholder',
+            ignoreFocusOut: true,
+        }
+
+        testInput = input.createInputBox({
+            options: options
```

Comment:
Nit: When the right-hand-side is the same as the left-hand-size, you can just do:

```typescript
{
   options
}
```